libraries/node-core-library/src/LockFile.ts

```diff
@@ -56,9 +56,9 @@ export class LockFile {
    * Returns the path to the lockfile, should it be created successfully.
    */
   public static getLockFilePath(resourceDir: string, resourceName: string, pid: number = process.pid): string {
-    if (!resourceName.match(/^[a-zA-Z]+$/)) {
+    if (!resourceName.match(/^[a-zA-Z0-9][a-zA-Z0-9-.]+[a-zA-Z0-9]$/)) {
```

Comment:
>. [](start = 52, length = 1)

For a filename like `"example-1.2.3.4.lock"` it's unclear whether "4" is a PID or not.  If you want to support periods, maybe we should use a different delimiter for the end, e.g. `"example-1.2.3#4.lock"` or `"example-1.2.3+4.lock"`?